test/show.jl

```diff
@@ -94,6 +94,20 @@ end
 @test_repr "(a / b) / (c / d / e)"
 @test_repr "(a == b == c) != (c == d < e)"
 
+# Exponentiation (>= operator_precedence(:^)) and unary operators
+@test_repr "(-1)^a"
+@test_repr "(-2.1)^-1"
+@test_repr "(-x)^a"
+@test_repr "(-a)^-1"
+@test_repr "(!x)↑!a"
+@test_repr "(!x).a"
+@test_repr "(!x)::a"
+
+# Complex
+
+@test iszero(eval(eval(parse(repr(:($(1 + 2im) - $(1 + 2im)))))))
```

Comment:
Oh! That makes sense but is a little subtle. I think `string` might go through slightly different code paths than `repr`, so either way. If you leave it like it is, maybe a brief comment to explain it would be worth adding for the benefit of future readers.